src/NuGet.Core/NuGet.PackageManagement/Projects/MSBuildNuGetProject.cs

```diff
@@ -619,8 +624,19 @@ private static bool IsAssemblyReference(string filePath)
             }
 
             // Assembly reference must have a .dll|.exe|.winmd extension and is not a resource assembly;
-            return !filePath.EndsWith(Constants.ResourceAssemblyExtension, StringComparison.OrdinalIgnoreCase) &&
-                   Constants.AssemblyReferencesExtensions.Contains(Path.GetExtension(filePath), StringComparer.OrdinalIgnoreCase);
+            if (Constants.AssemblyReferencesExtensions.Contains(Path.GetExtension(filePath), StringComparer.OrdinalIgnoreCase))
+            {
+                // Assembly name can be ended as ".resources" by itself. 
+                // We should check full name + resources.dll postfix here.
+                if (filePath.EndsWith(packageIdentity.Id + Constants.ResourceAssemblyExtension, StringComparison.OrdinalIgnoreCase))
+                {
+                    return false;
+                }
+
+                return true;
+            }
+
+            return false;                   
```

Comment:
The new code is pushed.  The same regex from the PackageReaderBase class is used to avoid code duplicates. A new unit test is added to cover this part of the functionality.  **TestMSBuildNuGetProjectDontIgnoreResourceAssembly**